### ISSUE TITLE
Mcast/Un-ucast exclusion problem

### ISSUE THREAD

**Issue by ScottDaniels** [att/vfd#115]
There is currently an issue when mcast is set to true, and un-ucast is set to false, and the reverse. The symptom is that setting one to false seems to cause both to be set to false.  I've verified that after updating the NIC with the desired values, reading the values back from the NIC show the expected settings, so this might not be a VFd bug, but an oddity of the underlying hardware.

**Comment by tkekan** [comment]
@ScottDaniels 
Just a note, we were seeing multicast drops even with mcast and uucast both set to true. 

Below are some sample jsons:

{"strip_stag": false, "macs": ["00:11:22:EF:AC:DD"], "name": "VFP1/e83261e0-35c6-4411-b8c3-356ea9a274ca", "link_status": "auto", "pciid": "0000:81:00.1", "allow_bcast": true, "vfid": 9, "mac_anti_spoof": true, "vlan_anti_spoof": true, "allow_mcast": true, "insert_stag": false, "vlans": [4001, 4002, 101, 102], "allow_un_ucast": true}

{"strip_stag": false, "macs": ["02:fb:f5:5c:dd:14"], "name": "Instance_admin-vmxc1_fpc0/fbf55cdd-14ca-49ba-84a6-330a1d076c8d", "link_status": "auto", "pciid": "0000:03:00.0", "allow_bcast": true, "vfid": 0, "mac_anti_spoof": true, "vlan_anti_spoof": true, "allow_mcast": true, "insert_stag": false, "vlans": [3501], "allow_un_ucast": true}

**Comment by ScottDaniels** [comment]
@tkekan Is this behaviour as before, which I think was a packet or two in and then nothing, or are you seeing more received packets but not 100%?

**Comment by tkekan** [comment]
@ScottDaniels 
Yes, this is seen since 1.14 deb package and I just pulled the latest code a couple of days back and I can see the same behavior. All multicast packets are blocked on the incoming side of VFD. Though, it is able to send out a multicast packet which gets successfully received by the remote end.

**Comment by ScottDaniels** [comment]
@tkekan  
Ok, you haven't tested with the code I pushed up this morning then.  That code has a change that may allow Rx of mcast when both mcast and un-ucast are set to true.

**Comment by tkekan** [comment]
@ScottDaniels : Great, Ill test that out then and update you!

**Comment by ScottDaniels** [comment]
@tkekan 
Much appreciated.  I thought you had pulled it which is why I hadn't mentioned it :)

**Comment by tkekan** [comment]
@ScottDaniels 
I no longer see the multicast packet drop. Your commit have fixed the issue. I'll keep you posted about any other issue seen and the testing is under progress.  
Thanks a lot :)

**Comment by ScottDaniels** [comment]
@tkekan 
Fantastic!  I'm leaving this open as we still have a related edge case as described in the initial text of this issue which we have to resolve in some shape.   Appreciate the quick testing!
